pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -99,8 +103,41 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
       return false;
     }
 
-    Comparable value = getValue(filterQueryTree.getValue().get(0), metadata.getDataType());
     PartitionFunction partitionFunction = metadata.getPartitionFunction();
+
+    if (operator == FilterOperator.EQUALITY) {
+      return pruneEquality(filterQueryTree, metadata, partitionRanges, partitionFunction);
+    }
+
+    return pruneIn(filterQueryTree, metadata, partitionRanges, partitionFunction);
+  }
+
+  private boolean pruneIn(FilterQueryTree filterQueryTree, ColumnMetadata metadata, List<IntRange> partitionRanges,
```

Comment:
Currently we are calling split for each segment for each in clause in DictionaryBasedInPredicateEvaluator in its call to getInRange. We are just moving all the evaluations to the pruner, so that a single thread does the evaluation freeing up the other threads to do more queries.